### DISCUSSION TITLE
What parameters needed to query by CURL or HTTPie ?

### DISCUSSION THREAD

**Discussion by coolbq** [alvarobartt/investpy#344] [Q&A]
Hi, I'm trying to build query strings to do download history data from investing.com. I want to query by CURL or HTTPie. But it returns error this way. Does investing.com need any special parameters in the query?
I also test it in the Postman, but returns the same error. The query string is like this one "**https://www.investing.com/instruments/HistoricalDataAjax?curr_id=1114321&smlID=27721383&header=MRNA Historical Data&st_date=02/01/2021&end_date=04/27/2021&interval_sec=Daily&sort_col=date&sort_ord=DESC&action=historical_data**"

Do I miss anything? Or it is not possible to query by CURL or HTTPie ?

**Comment by alvarobartt** [comment]
Hi @coolbq, AFAIK you can cURL it, but you need to specify not just the parameters, but also the headers and so on, for a quick reference, check this piece of code from `investpy` and try to replicate it using cURL. If it works let us know how you did it! 👍🏻 

https://github.com/alvarobartt/investpy/blob/26a61cafa9302ecc4eeffa6bf72fc5a0cd8c8ff8/investpy/stocks.py#L273-L295

For your concrete example, the header value should be `"MRNA Historical Data"`, as `MRNA` is the symbol.
